CRM/Core/OptionGroup.php

```diff
@@ -676,10 +676,23 @@ public static function flush($name, $params = array()) {
     );
   }
 
+  /**
+   * Flush all the places where option values are cached.
+   */
   public static function flushAll() {
+    CRM_Core_OptionGroup::flushCachedGreetings();
     self::$_values = array();
     self::$_cache = array();
     CRM_Utils_Cache::singleton()->flush();
   }
 
+  /**
+   * Flush any places where greeting strings are cached.
+   */
+  public static function flushCachedGreetings() {
+    if (isset(Civi::$statics['CRM_Core_PseudoConstant']['greeting'])) {
+      unset(Civi::$statics['CRM_Core_PseudoConstant']['greeting']);
+    }
+  }
```

Comment:
This is a strange oddball. Why is this cache variable being stored in `Civi::$statics` instead of in `CRM_Core_PseudoConstant::$cache`? If it was in the latter then no special function would be needed and `CRM_Core_PseudoConstant::flush()` would take care of it.

Reply:
We've been consistently moving towards storing statics in Civi::Statics - I can go a different way if you want - but there are a lot of places where that has been done. Note that CiviStatics IS flushed between test runs - but there is sometimes a need to flush a value once it has been altered. We don't normally flush all PseudoConstants in order to achieve that since there is still a performance benefit in keeping most of our cached variables.

Reply:
I still think putting it in `CRM_Core_PseudoConstant::$cache` would be better because we then wouldn't need this function. If we followed the pattern you're establishing here of a clear function for every variable, this file would get quite long & unwieldy.

Reply:
I removed this function because I added in to unset Civi::statics[__CLASS__] in the CRM_Core_PseudoConstant function. There are already a few functions in that class using the Civi::statics.